build.js

```diff
@@ -194,3 +196,29 @@ function prependToTestName(prefix) {
     };
 }
 
+/**
+ * Verifies that no tests have duplicate names.
+ * @param allTests - The list of all tests.
+ */
+function checkDuplicateNames(allTests) {
+    const names = new Set();
+    const duplicates = [];
+
+    for (const test of allTests) {
+        const oldSize = names.size;
+        const name = test.name;
+        names.add(name);
+
+        // If size remained the same, a duplicate was encountered
+        if (oldSize === names.size) {
+            duplicates.push(name);
+        }
+    }
+
+    if (duplicates.length > 0) {
+        duplicates.sort();
+        const prefix = "\n  ";
+        console.error("Error: Duplicate test names:" + prefix + duplicates.join(prefix));
+        process.exit(1);
```

Comment:
Do we really want to fail the build or just emit a warning? Not suggesting, just asking.

Reply:
If we do fail, it makes sense to pick a different exit code and define them at the top of the script.

Reply:
At least there should be some flag then to let CI fail for duplicate test names. But I think it might be best to always fail. Duplicate names will just be confusing.

Reply:
GH Actions doesn't support warnings (as far as I know).  At least, they're not visible on this "conversation" page.  You have to go into the file changes to see anything.  It shows up like this:

![image](https://github.com/user-attachments/assets/e70c9dc4-62e3-48b7-9b53-0b83e02e7ea8)

I think I'd prefer the failure.

Reply:
I'm fine either way, just let's use another exit code, `1` is used here already https://github.com/jsonpath-standard/jsonpath-compliance-test-suite/blob/733fe526ceccb183ed0e3397340f2ee9feec0d67/build.js#L71

Reply:
@f3ath I'm curious of your reason why.  Does it make a different in local development?

Reply:
Just future-proofing in case we introduce more checks and/or want to act differently depending on why it failed.